google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/features/ObjectApi.java

```diff
@@ -199,7 +198,7 @@ GoogleCloudStorageObject simpleUpload(@PathParam("bucket") String bucketName, @H
    @Named("Object:delete")
    @DELETE
    @Path("storage/v1/b/{bucket}/o/{object}")
-   @Fallback(TrueOnNotFoundOr404.class)
+   @Fallback(FalseOnNotFoundOr404.class)
```

Comment:
I've extended the "return-false-on-404" behavior to `deleteBucket()` as well.
And added a couple of unit tests.